pkg/virt-v2v/customize/scripts/rhel/run/network_config_util.sh

```diff
@@ -274,7 +274,22 @@ udev_from_ifquery() {
         fi
 
         # Create the udev rule based on the extracted MAC address and interface name
-        echo "SUBSYSTEM==\"net\",ACTION==\"add\",ATTR{address}==\"$(remove_quotes "$S_HW")\",NAME=\"$(remove_quotes "$interface_name")\""
+        echo "SUBSYSTEM==\"net\",ACTION==\"add\",ATTR{address}==\"$(remove_quotes "$S_HW")\",NAME=\"temp_$(remove_quotes "$interface_name")\""
+    done
+}
+
+
+# Create final udev rules based on the temporary udev rules
+udev_final_from_temp_rules() {
+    # Read the temporary udev rules file line by line
+    cat "$UDEV_RULES_FILE" | while read line;
+    do
+        # Extract the temporary name and final name
+        TEMP_NAME=$(echo "$line" | grep -o 'NAME="temp_[^"]*"' | cut -d'"' -f2)
+        FINAL_NAME=$(echo "$TEMP_NAME" | sed 's/^temp_//')
+
+        # Generate final udev rule by replacing the temporary name with the final name in the original line
+        echo "$(echo "$line" | sed "s/temp_$FINAL_NAME/$FINAL_NAME/")"
```

Comment:
This could be replaced by 
```bash
    input=$(cat)
    echo "$input"
    echo "$input" | sed "s/temp_//"
```
and append it to the pipe
```bash
   {
        udev_from_ifcfg
        udev_from_nm
        udev_from_netplan
        udev_from_ifquery
    } | check_dupe_hws | udev_final_from_temp_rules > "$UDEV_RULES_FILE" 2>/dev/null
```